controllers/cluster_controller.go

```diff
@@ -287,3 +296,57 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileClusterStatus(ctx context.Context, cluster *clusterv1.Cluster) error {
```

Comment:
I'm wondering if this should be renamed to something like reconcileControlPlaneInitialized, since there are other Status fields that are reconciled elsewhere currently?

Reply:
+1